game/gamestate.go

```diff
@@ -147,3 +308,149 @@ func (gs *GameState) AreAdjacent(cantonID1, cantonID2 int) bool {
 	}
 	return false
 }
+
+func (gs *GameState) Delta() map[string]any {

```

Comment:
need to record and return the static board positions difference, so the outcomes of an attack can be compared easily.